src/MongoDB.EntityFrameworkCore/Infrastructure/MongoModelValidator.cs

```diff
@@ -261,13 +261,7 @@ private static void ValidateEntityQueryableEncryption(
             }
 
             var dataKeyId = property.GetEncryptionDataKeyId();
-            if (dataKeyId == null)
```

Comment:
Allowing empty data keys if they happened to get here somehow as the shared lib throws albeit a less helpful error. Is a bit difficult to actually get here unless they set annotations manually using the direct annotation api.